File: server/php/src/Application/Domain/Payments/PaymentDetails/ThreeDSecureResults.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class ThreeDSecureResults
{
    public ?string $acsTransactionId = null;
    public ?string $appliedExemption = null;
    public ?string $authenticationStatus = null;
    public ?string $cavv = null;
    public ?string $challengeIndicator = null;
    public ?string $dsTransactionId = null;
    public ?string $eci = null;
    public ?string $exemptionEngineFlow = null;
    public ?string $flow = null;
    public ?string $liability = null;
    public ?string $schemeEci = null;
    public ?string $version = null;
    public ?string $xid = null;

    public function toArray(): array
    {
        return [
            'acsTransactionId' => $this->acsTransactionId,
            'appliedExemption' => $this->appliedExemption,
            'authenticationStatus' => $this->authenticationStatus,
            'cavv' => $this->cavv,
            'challengeIndicator' => $this->challengeIndicator,
            'dsTransactionId' => $this->dsTransactionId,
            'eci' => $this->eci,
            'exemptionEngineFlow' => $this->exemptionEngineFlow,
            'flow' => $this->flow,
            'liability' => $this->liability,
            'schemeEci' => $this->schemeEci,
            'version' => $this->version,
            'xid' => $this->xid,
        ];
    }
}

File: server/python/app/application/domain/payments/payment_details/three_d_secure_results.py
from dataclasses import dataclass
from typing import Optional

@dataclass
class ThreeDSecureResults:
    acs_transaction_id: Optional[str] = None
    applied_exemption: Optional[str] = None
    authentication_status: Optional[str] = None
    cavv: Optional[str] = None
    challenge_indicator: Optional[str] = None
    ds_transaction_id: Optional[str] = None
    eci: Optional[str] = None
    exemption_engine_flow: Optional[str] = None
    flow: Optional[str] = None
    liability: Optional[str] = None
    scheme_eci: Optional[str] = None
    version: Optional[str] = None
    xid: Optional[str] = None